lib/cartopy/mpl/gridliner.py

```diff
@@ -276,17 +342,23 @@ def _add_gridline_label(self, value, axis, upper_end):
         # Make a 'blended' transform for label text positioning.
         # One coord is geographic, and the other a plain Axes
         # coordinate with an appropriate offset.
-        label_transform = mtrans.blended_transform_factory(
-            x_transform=tr_x, y_transform=tr_y)
+        if other_value is None:
+            label_transform = mtrans.blended_transform_factory(
+                x_transform=tr_x, y_transform=tr_y)
+            alpha = 1
+        else:
+            label_transform = cartopy.crs.PlateCarree()
+            alpha = 0.7
```

Comment:
I'm not sure if we want to be changing transparency. Best would be if the line were cut under the label, like contour lines, but this is somewhat more difficult to do. But maybe using the contour machinery would help.